include/libopencm3/stm32/common/usart_common_fl_0.h

```diff
@@ -0,0 +1,296 @@
+#define USART_ICR_CTSCF			(1 << 9)
+#define USART_ICR_LBDCF			(1 << 8)
+#define USART_ICR_TCCF			(1 << 6)
+#define USART_ICR_IDLECF		(1 << 4)
+#define USART_ICR_ORECF			(1 << 3)
+#define USART_ICR_NCF			(1 << 2)
+#define USART_ICR_FECF			(1 << 1)
+#define USART_ICR_PECF			(1 << 0)
+
+/*****************************************************************************/
+/* API definitions                                                           */
+/*****************************************************************************/
+
+#define USART_PARITY			(USART_CR1_PCE | USART_CR1_PS)
+#define USART_PARITY_NONE		(0)
+#define USART_PARITY_EVEN		(USART_CR1_PCE)
+#define USART_PARITY_ODD		(USART_CR1_PCE | USART_CR1_PS)
+
+#define USART_MODE			(USART_CR1_TE | USART_CR1_RE)
+#define USART_MODE_NONE			(0)
+#define USART_MODE_RX			(USART_CR1_RE)
+#define USART_MODE_TX			(USART_CR1_TE)
+#define USART_MODE_TX_RX		(USART_CR1_TE | USART_CR1_RE)
+
+#define USART_FLOWCONTROL		(USART_CR3_RTSE | USART_CR3_CTSE)
+#define USART_FLOWCONTROL_NONE		(0)
+#define USART_FLOWCONTROL_RTS		(USART_CR3_RTSE)
+#define USART_FLOWCONTROL_CTS		(USART_CR3_CTSE)
+#define USART_FLOWCONTROL_RTS_CTS	(USART_CR3_RTSE | USART_CR3_CTSE)
+
+/*****************************************************************************/
+/* API Functions                                                             */
+/*****************************************************************************/
+
+BEGIN_DECLS
+
+void usart_set_baudrate(uint32_t usart, uint32_t baud);
+void usart_set_databits(uint32_t usart, uint32_t bits);
+void usart_set_stopbits(uint32_t usart, uint32_t stopbits);
+void usart_set_parity(uint32_t usart, uint32_t parity);
+void usart_set_mode(uint32_t usart, uint32_t mode);
+void usart_set_flow_control(uint32_t usart, uint32_t flowcontrol);
+void usart_enable(uint32_t usart);
+void usart_disable(uint32_t usart);
+void usart_send(uint32_t usart, uint16_t data);
+uint16_t usart_recv(uint32_t usart);
+void usart_wait_send_ready(uint32_t usart);
+void usart_wait_recv_ready(uint32_t usart);
+void usart_send_blocking(uint32_t usart, uint16_t data);
+uint16_t usart_recv_blocking(uint32_t usart);
+void usart_enable_rx_dma(uint32_t usart);
+void usart_disable_rx_dma(uint32_t usart);
+void usart_enable_tx_dma(uint32_t usart);
+void usart_disable_tx_dma(uint32_t usart);
+void usart_enable_rx_interrupt(uint32_t usart);
+void usart_disable_rx_interrupt(uint32_t usart);
+void usart_enable_tx_interrupt(uint32_t usart);
+void usart_disable_tx_interrupt(uint32_t usart);
+void usart_enable_error_interrupt(uint32_t usart);
+void usart_disable_error_interrupt(uint32_t usart);
+bool usart_get_flag(uint32_t usart, uint32_t flag);
+bool usart_get_interrupt_source(uint32_t usart, uint32_t flag);
+uint32_t usart_get_clock(uint32_t usart);
+void usart_set_autobaudrate(uint32_t usart, int mode);
+
+END_DECLS
+
+#endif
+/** @cond */
+#else
+#warning "usart_common_fl_0.h should not be included directly, only via usart.h"
```

Comment:
this should just be an #error, as we did in https://github.com/libopencm3/libopencm3/commit/721f3d74eb20bbbbe7438c98adece47adc7e7a0b